deepspeed/runtime/pipe/engine.py

```diff
@@ -49,14 +49,15 @@ class PipelineEngine(DeepSpeedEngine):
     This engine is created by ``deepspeed.initialize()`` when a :class:`PipelineModule`
     is provided.
     """
-    def __init__(self, *super_args, **super_kwargs):
+    def __init__(self, has_bool_tensors=False, *super_args, **super_kwargs):
```

Comment:
I would prefer for this PP changes to be a separate PR, as it seems fairly independent of the activation checkpointing changes. Also, I am more comfortable reviewing the activation checkpointing changes but will need other team members to review the PP changes. Thanks.

Reply:
good idea. Should I close this PR and open a new one?

Reply:
I re-opend https://github.com/microsoft/DeepSpeed/pull/1376 . 
I will remove contiguous checkpointing parts in this PR. 
Let's discuss above PR.